### ISSUE TITLE
csv api query returns 404 error

### ISSUE THREAD

**Issue by charlieeatonphd** [RTICWDT/college-scorecard#1846]
When you change an API query from .json such as:

https://api.data.gov/ed/collegescorecard/v1/schools.json?school.degrees_awarded.predominant=2,3&fields=id,school.name,2013.student.size&api_key=n3vzgGatzVDfUpe5jnFyLYhYriTdh1GhrFpw8ADG

to .csv,  such as:

https://api.data.gov/ed/collegescorecard/v1/schools.csv?school.degrees_awarded.predominant=2,3&fields=id,school.name,2013.student.size&api_key=n3vzgGatzVDfUpe5jnFyLYhYriTdh1GhrFpw8ADG

you get the following error:

> Oops! An Error Occurred
> 
> The server returned a "404 Not Found".
> 
> Something is broken. Please let us know what you were doing when this error occurred. We will fix it as soon as possible. Sorry for any inconvenience caused.

**Comment by brownpl** [comment]
Thanks for reporting. I'll check into it and hopefully have a fix for it later this week.

**Comment by brownpl** [comment]
I've added a csv option. It is currently posted to our beta endpoint - is that something you could use for testing? I am waiting to see if any other issues come in that I can batch with this update, and aim to update production this weekend. 

The beta endpoint is https://api.data.gov/ed/collegescorecard/beta/v1/schools - your existing keys should work fine.

**Comment by charlieeatonphd** [comment]
I can try that later today.

You all had a CSV option that worked until Sunday April 4. Something done
on Sunday broke it.
Best,
Charlie
[image: photo]
*Charlie Eaton (he/him)*
Assistant Professor of Sociology, UC Merced
Affiliated Faculty, UC Berkeley Center for Studies in Higher Education
web: faculty.ucmerced.edu/ceaton2/
<https://twitter.com/charlieeatonphd>
<https://www.facebook.com/eaton.charlie>

**Comment by brownpl** [comment]
Yes, there was a big API update on Sunday. The details are in the [change log](https://collegescorecard.ed.gov/data/changelog/).

**Comment by brownpl** [comment]
The .csv functionality should now be available for you on the main endpoint.

**Comment by charlieeatonphd** [comment]
Thx!
Best,
Charlie
[image: photo]
*Charlie Eaton (he/him)*
Assistant Professor of Sociology, UC Merced
Affiliated Faculty, UC Berkeley Center for Studies in Higher Education
web: faculty.ucmerced.edu/ceaton2/
<https://twitter.com/charlieeatonphd>
<https://www.facebook.com/eaton.charlie>

**Comment by charlieeatonphd** [comment]
Hi. This still doesn't work. Neither

https://api.data.gov/ed/collegescorecard/beta/v1/schools.csv?&fields=id,school.name,2016.repayment.repayment_cohort.3_year_declining_balance,2016.repayment.3_yr_repayment_suppressed.overall,2016.repayment.3_yr_repayment.overall&_per_page=100&_page=0&api_key=[MYKEY]

nor

https://api.data.gov/ed/collegescorecard/v1/schools.csv?&fields=id,school.name,2016.repayment.repayment_cohort.3_year_declining_balance,2016.repayment.3_yr_repayment_suppressed.overall,2016.repayment.3_yr_repayment.overall&_per_page=100&_page=0&api_key=[MYKEY]

Please advise.

**Comment by brownpl** [comment]
A few initial thoughts. The first one is no longer available, our beta endpoint has been removed and all the functionality that was previously there should be available on production (the second link). 

I just ran a few tests with your second one - for some reason, I am getting an invalid response in Chrome, but if I run it in Safari (I'm on a mac) I am getting a CSV download. Are you accessing this via the browser or by some other method? If it is the browser, a quick stopgap solution could be to try a different one. In the meantime, I'll see what I can figure out around why Chrome is having issues, and do a pass over others to ensure they are working as expected, and I'll respond here when I sort it out.

**Comment by brownpl** [comment]
After the data update last week, the export is appearing to work now as expected. I am not able to replicate the same issues with the invalid response. Please give it a shot and let me know if it remains an issue for you.
